src/kompos/hierarchical/composition_config_generator.py

```diff
@@ -55,16 +55,29 @@ def split_path(value, separator='='):
     return [value, ""]
 
 
+# Get hiera config path
 def get_config_path(path_prefix, composition):
     prefix = os.path.join(path_prefix, '')
-    return path_prefix if "composition=" in path_prefix else "{}composition={}".format(
-        prefix, composition)
+    if "composition=" in path_prefix:
+        return path_prefix
+    else:
+        return "{}composition={}".format(prefix, composition)
 
 
-def get_composition_path(path_prefix, composition):
+# Get target composition path
+def get_composition_path(path_prefix, composition, raw_config):
     prefix = os.path.join(path_prefix, '')
-    return path_prefix if composition in path_prefix else "{}{}/".format(
-        prefix, composition)
+    if composition not in path_prefix and composition == "custom":
+        try:
+            custom_composition = raw_config["custom"]["type"]
+            logger.info("Appending custom composition: %s", custom_composition)
+            return "{}{}/{}/".format(prefix, composition, custom_composition)
+        except CustumCompositionNotFound:
```

Comment:
Typo s/Custum/Custom

Reply:
This exception doesn't seem to exist, no it will never be caught. Also none of the commands in the try block throw it.